physics/discrete_trajectory.hpp

```diff
@@ -182,20 +182,56 @@ class DiscreteTrajectory : public Forkable<DiscreteTrajectory<Frame>,
   std::int64_t timeline_size() const override;
 
  private:
-  struct Downsampling {
+  class Downsampling {
+   public:
+    Downsampling(std::int64_t max_dense_intervals,
+                 Length tolerance,
+                 TimelineConstIterator start_of_dense_timeline,
+                 Timeline const& timeline);
+
+    TimelineConstIterator start_of_dense_timeline() const;
+    // |start_of_dense_timeline()->first|, for readability.
+    Instant const& first_dense_time() const;
+    // Keeps |dense_intervals_| consistent with the new
+    // |start_of_dense_timeline_|.
+    void set_start_of_dense_timeline(TimelineConstIterator value,
```

Comment:
Should probably be called `SetStartOfDenseTimeline` as it is expensive.

Reply:
Done.